ecommerce/extensions/offer/utils.py

```
@@ -99,3 +100,69 @@ def render_email_confirmation_if_required(request, offer, product):
                 'user_email': request.user and request.user.email,
             }
         )
+
+
+def send_assigned_offer_email(
+        template, offer_assignment_id, learner_email, code, enrollment_url, code_usage_count, code_expiration_date):
+    """
+    Arguments:
+        *template*
+            The email template with placeholders that will receive the following tokens
+        *offer_assignment_id*
+            Primary key of the entry in the offer_assignment model.
+        *learner_email*
+            Email of the customer who will receive the code.
+        *code*
+            Code for the user.
+        *enrollment_url*
+            URL for the user.
+        *code_usage_count*
+            Number of times the code can be redeemed.
+        *code_expiration_date*
+            Date till code is valid.
+
+    Returns:
+         True when successful or False in case of any exception
```

This seems like an unusual choice for a function like this: why not let the exception raise up to the caller, and have them deal with it?